src/main/java/com/iopipe/generic/EntryPoint.java

```diff
@@ -577,5 +505,271 @@ else if (pn == 0)
+			if (used == null)
+				return null;
+			
+			// Allow us to call this method without performing any access checks
+			boolean access = used.isAccessible();
+			if (!access)
+				try
+				{
+					used.setAccessible(true);
+				}
+				catch (SecurityException e)
+				{
+				}
+			
+			// Get method handle from it, assuming the previous call worked
+			MethodHandle basehandle;
+			try
+			{
+				basehandle = MethodHandles.lookup().unreflect(used);
+			}
+			catch (IllegalAccessException e)
+			{
+				throw new InvalidEntryPointException("Could not access the " +
+					"generic entry point method.", e);
+			}
+			
+			// If this was not accessible, then we would have tried to make it so
+			// so just revert access to it
+			if (!access)
+				try
+				{
+					used.setAccessible(false);
+				}
+				catch (SecurityException e)
+				{
+				}
+			
+			// Determine the types of arguments this takes, this will require
+			// parameterized type resolution as required
+			// The original parameters will always be used as the base
+			Type[] xargs = used.getGenericParameterTypes();
+			for (int i = 0; i < numargs; i++)
+			{
+				Type t = __Target__.__resolve(clstack, xargs[i]);
+				if (t != null)
+					xargs[i] = t;
+			}
+			
+			// Initialize target information
+			return new __Target__(basehandle, xargs,
+				((used.getModifiers() & Modifier.STATIC) != 0));
+		}
+		
+		/**
+		 * Resolves the given type variable for a class.
+		 *
+		 * @param __cls The class stack.
+		 * @param __t The type to resolve.
+		 * @throws NullPointerException On null arguments.
+		 * @since 2018/09/05
+		 */
+		static Type __resolve(Deque<Class<?>> __cls, Type __t)
+			throws NullPointerException
+		{
+			if (__cls == null || __t == null)
+				throw new NullPointerException();
+			
+			// If it is not a type variable, just return self
+			if (!(__t instanceof TypeVariable))
+				return __t;
+			
+			// Make a copy of the stack
+			Deque<Class<?>> stackcopy = new ArrayDeque<>(__cls);
+			
+			TypeVariable tv = (TypeVariable)__t;
+			String tvname = tv.getName();
+			
+			// The class that declares this type parameter was the
+			// one the method was found in (it will be at the top)
+			Class<?> pivotclass = stackcopy.pop();
+			
+			// Determine which index our type variable is in, we
+			// need to look in the super class
+			TypeVariable<?>[] pivotvars = pivotclass.getTypeParameters();
+			int pivotdx = -1;
+			for (int j = 0, n = pivotvars.length; j < n; j++)
+				if (tvname.equals(pivotvars[j].getName()))
+				{
+					pivotdx = j;
+					break;
+				}
+			
+			// This should not happen normally (unless the class
+			// was modified to break it or we are trying to initialize
+			// a non-static inner class which cannot be initialized
+			// anyway)
+			if (pivotdx < 0)
+				return __t;
+			
+			// Peek the current class
+			Class<?> upperclass = stackcopy.peek();
+			
+			// Initializing a class which is just a type variable
+			if (upperclass == null)
+				return __t;
```

Comment:
This needs to be changed to `return ObjectTranslator.__unbound(tv)` so at least the base class matches even though we do not really know what it is. Example from #163 `I` will be translated to `StringGetter<I>`.

Reply:
Test object `ParameterA<A, B>` should have its bounds changed to  `A extends SomeClass ` that does something.